javascript/main.js

```diff
@@ -0,0 +1,154 @@
+// Create one global variable (e.g. CarLot) and use the IIFE pattern to augment it two times in separate JavaScript files.
+// The first IIFE 
+// When your page first loads, XHR loads the contents of the JSON file, and parse them into a native JavaScript object. (see line 20)
+"use strict";
+
+var carLot = (function(carLot) {
+
+  var inventoryRequest = "";
+  var inventory = [];
+  
+
+  ////////////////////////////
+  ///JSON LOADER FUNCTIONSET
+  
+
+  //a public function that loads the inventory.json file  
+  carLot.loadInventory = function() {
+    inventoryRequest = new XMLHttpRequest();
+    inventoryRequest.open("GET", "data/inventory.json");
+    inventoryRequest.send();
+    inventoryRequest.addEventListener("load", carLot.setInventory);
+    inventoryRequest.addEventListener("error", carLot.failedToLoad);  
+  };
+
+  //function to load if the json fails. 
+  carLot.failedToLoad = function() {
+    console.log("JSON failed to load" );
```

Comment:
Since this is an error, you can replace the ```log``` with ```error``` to make it nice and red and angry.